Src/GoogleApis.Auth.DotNet4.Tests/OAuth2/GoogleCredentialTests.cs

```diff
@@ -67,7 +70,40 @@ public void FromStream_ServiceAccountCredential()
         {
             var stream = new MemoryStream(Encoding.UTF8.GetBytes(DummyServiceAccountCredentialFileContents));
             var credential = GoogleCredential.FromStream(stream);
+            Assert.IsInstanceOf(typeof(JwtServiceAccountCredential), credential.UnderlyingCredential);
             Assert.IsTrue(credential.IsCreateScopedRequired);
         }
+
+        /// <summary>
+        /// Creates service account credential from stream, obtains a JWT token
+        /// from the credential and checks the access token is well-formed.
+        /// </summary>
+        /// <returns></returns>
+        [Test]
+        public async Task FromStream_ServiceAccountCredential_GetJwtAccessToken()
+        {
+            var stream = new MemoryStream(Encoding.UTF8.GetBytes(DummyServiceAccountCredentialFileContents));
+            var credential = GoogleCredential.FromStream(stream);
+
+            // Without adding scopes, the credential should be generating JWT scopes.
+            string accessToken = await (credential as ITokenAccess).GetAccessTokenForRequestAsync(DummyAuthUri);
```

Comment:
I think this is clearer for the user.

Imagine you obtain a GoogleCredential object using GetApplicationDefaultCredential().   Now what you want to do is maybe set scopes to it and maybe pass it to BaseClientService.Initializer or to a gRPC specific auth class.

In none of those case you actually need to call the methods provided IConfigurableHttpClient initializer or ITokenAccess - you just use the fact that the credential object inherits from those interfaces, but you don't wanna call the methods provided by it. On the other hand, the BaseClientService will see the credential as an instance of IConfigurableHttpClient - so it will see the explictly implemented methods right away.

So, I think explicitly inheriting from an interface is what we should do here - it hides the methods the user will never need to call himself from him (he can still call them if he chooses to - but they won't be visible in the common use case), on the other hand, the internal code that will need to call those can call them with no extra effort.